cms/src/components/content/ContentForm.spec.ts

```diff
@@ -256,6 +256,41 @@ describe("ContentForm", () => {
         expect(wrapper.text()).toContain("Slug:updated-title");
     });
 
+    it("displays a text input when the slug edit button is pressed", async () => {
+        const wrapper = mount(ContentForm, {
+            props: {
+                post: mockPost,
+                content: mockEnglishContent,
+            },
+        });
+
+        await wrapper.find("button[name='editSlugButton']").trigger("click");
+
+        await flushPromises();
+        const input = wrapper.find("input[name='slug']");
+        expect(input.isVisible()).toBe(true);
+    });
+
+    it("saves the new slug when the text input loses focus and hides the input", async () => {
+        const wrapper = mount(ContentForm, {
+            props: {
+                post: mockPost,
+                content: mockEnglishContent,
+            },
+        });
+        const input = wrapper.find("input[name='slug']");
+        const slug = wrapper.find("span[name='slugSpan']");
```

Comment:
As discussed, this makes the test very dependent on the HTML layout, so even small changes like changing the `span` to a `div` will make the test failing (and I think they should only fail if you actually change the functionality to a certain degree). It does make the tests a little nicer to read so I understand if you want to keep it. I would just make two changes in any case:

- remove the `span` (the `name` is already unique enough)
- Change `name` to `data-test`, which I've used in other tests for that purpose. `name` is not valid on a `span`, but `data-*` elements are. This comment holds for the every test except the `input`s where `name` makes sense.